File: c/fibonacci/fibonacci.c
#include <stdio.h>

// To run, compile it with gcc and run the executable passing: ./a.out n, n being the number used in the fibonacci

int fib(int n) { return (n==0 || n==1) ? 1 : (fib(n-1) + fib(n-2)); }

int main(int argc, char *argv[]) {
  printf("Fibonacci: %d\n", fib((int) *argv[1] - '0'));
}

File: rust/fibonacci/fibonacci.rs
fn main() {
    let num: u32 = 20;
    println!("The {} number in the fibonacci sequence is: {}", num, fib(num));
}

fn fib(x: u32) -> u32 { match x {0 => 1, 1 => 1, _ => fib(x - 1) + fib(x - 2),}}
